deluge/core/alertmanager.py

```diff
@@ -41,13 +41,43 @@
 
 """
 
-from twisted.internet import reactor
+import threading
+
+from twisted.internet import reactor, threads
 
 import deluge.component as component
 from deluge._libtorrent import lt
 
 from deluge.log import LOG as log
 
+class AlertPoller(object):
+
+    def __init__(self, alert_manager):
+        self.alert_manager = alert_manager
+        self.running = False
+
+    def maybeUpdate(self):
+        if self.running:
+            self.alert_manager.update()
+
+    def run(self):
+        while self.running:
+            try:
+                alert = self.alert_manager.session.wait_for_alert(1000)
+            except:
```

Comment:
You should probably at least log the exception here since it should rarely (if ever) occur and it would be good for debugging issues.

Reply:
I think I already do this.